docs/datastorage/capabilities.md

```diff
@@ -0,0 +1,176 @@
+
+Using an Existing Capability
+----------------------------
+
+This most notably applies to the Item Handler capability added by Forge, which replaces the vanilla IInventory and ISidedInventory with an automation-friendly system.
+
+As mentioned earlier, TileEntities, Entities, and ItemStacks implement the capability provider feature, through the `ICapabilityProvider` interface. This interface adds two methods, `hasCapability` and `getCapability`, which can be used to query the capabilities present in the objects.
+
+In order to obtain a capability, you will need to refer it by its unique instance. In the case of the Item Handler, this capability is primarily stored in `CapabilityItemHandler.ITEM_HANDLER_CAPABILITY`, but it is possible to get an instance reference by using the `@CapabilityInject` annotation. This annotation can also be used on a method in order to get notified when a capability is registered, so that certain features can be enabled conditionally. Please refer to the javadocs for this annotation for details on usage.
+
+Both the `hasCapability` and `getCapability` methods have a second parameter, of type EnumFacing, which can be used in the TileEntities to request the specific instance for that one face. If passed `null`, a general capability instance will be requested instead. The return type of `getCapability` will correspond to the type declared in the capability passed to the method. For the Item Handler capability, this is indeed `IItemHandler`.
+
+Exposing a Capability
+---------------------
+
+In order to expose a capability, you will first need an instance of the underlying capability type. There's two ways to obtain such an instance, through the Capability itself, or by explicitly instantiating an implementation of it.
+
+The first method is designed to use a default implementation, if those default values are useful for you. In the case of the Item Handler capability, the default implementation will expose a single slot inventory, which is most probably not what you want.
+
+The second method can be used to customize the existing implementations, or to implement your own. In the case of `IItemHandler`, the default implementation uses the `ItemStackHandler` class, which has an optional argument in the constructor, to specify a number of slots.
+
+```Java
+// A dispenser-like block could use something like this
+IItemHandler inventory = new ItemStackHandler(9);
+```
+
+Once you have your own instance of the capability interface, you will want to notify users of the capability system that you expose this capability. This is done by overriding the `hasCapability` method, and comparing the instance with the capability you are exposing. If your machine has different slots based on which side is being queried, you can test this with the `facing` parameter. For entities and ItemStacks, this parameter can be ignored. Don't forget to fall back to `super`, otherwise the attached capabilities will stop working.
+
```

Comment:
The facing can be ignored by any implementors but if your system is side aware, it should take this into account.
For example in Players:
IItemHandler COULD be expanded to be side aware and have say, UP == Helmet, DOWN = Boots, ANY_SIDE = chest/sword/legs.
We won't be doing that in forge, but that is just an example of where sided-ness may be import in entities.
Same goes for item stacks, Example:
A mod wants to make furnaces function in a inventory. So on the 2d grid, Sides=output, Top = Input, Bottom = Fuel.

It is also worth it to note that get/hasCapability is designed to be fast, use only If statements when you can, don't rely on maps and the like.  has is also meant to assist in LAZY loading of the capability. If that is the implementer's intention.